lib/components/joystick/joystick_component.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:ui';
+
+import 'package:flame/components/component.dart';
+import 'package:flame/components/joystick/joystick_action.dart';
+import 'package:flame/components/joystick/joystick_directional.dart';
+import 'package:flame/components/joystick/joystick_events.dart';
+import 'package:flame/components/mixins/has_game_ref.dart';
+import 'package:flame/game/base_game.dart';
+import 'package:flame/gestures.dart';
+
+abstract class JoystickListener {
```

Comment:
Since this is not inherited, I think it would be nicer if this was a mixin, what do you think?